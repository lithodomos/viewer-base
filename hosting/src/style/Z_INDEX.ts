const HEADER = 999;
const LOADING_SCREEN = 999;
const MENU = LOADING_SCREEN - 100;
const OVERLAY_UI = MENU - 100;

// inside tour
const STREET_VIEW = OVERLAY_UI - 300;
const IMAGE_VIEW = STREET_VIEW;
const PANORAMA = STREET_VIEW;

const PANORAMA_LOADING = LOADING_SCREEN;

const VIEWER = PANORAMA - 10;

const FULLSCREEN_VIDEO = PANORAMA;

const MODAL =
  Math.max(
    HEADER,
    OVERLAY_UI,
    STREET_VIEW,
    IMAGE_VIEW,
    PANORAMA,
    VIEWER,
    OVERLAY_UI,
    MENU,
    FULLSCREEN_VIDEO
  ) + 1;

const REACT_TOAST_CONTAINER = MODAL + 1;

export const Z_INDEX = {
  OVERLAY_UI,
  VIEWER,
  PANORAMA,
  STREET_VIEW,
  IMAGE_VIEW,
  HEADER,
  LOADING_SCREEN,
  PANORAMA_LOADING,
  MENU,
  FULLSCREEN_VIDEO,
  MODAL,
  REACT_TOAST_CONTAINER,
};
