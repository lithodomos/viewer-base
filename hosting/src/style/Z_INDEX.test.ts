import { Z_INDEX } from './Z_INDEX';

test('it matches the snapshot', () => {
  expect(Z_INDEX).toMatchSnapshot();
});
