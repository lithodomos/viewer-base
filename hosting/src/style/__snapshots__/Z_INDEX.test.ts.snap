// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches the snapshot 1`] = `
Object {
  "FULLSCREEN_VIDEO": 499,
  "HEADER": 999,
  "IMAGE_VIEW": 499,
  "LOADING_SCREEN": 999,
  "MENU": 899,
  "MODAL": 1000,
  "OVERLAY_UI": 799,
  "PANORAMA": 499,
  "PANORAMA_LOADING": 999,
  "REACT_TOAST_CONTAINER": 1001,
  "STREET_VIEW": 499,
  "VIEWER": 489,
}
`;
